src/schema/types.ts

```diff
@@ -80,3 +90,15 @@ export const isItemsArray = (
  */
 
 export const isRegex = (regexp: any): regexp is RegExp => regexp;
+
+export const hasAnyOf = (value: JSONSchema7): value is JSONSchema7 & { anyOf: JSONSchema7Definition[]} =>
```

Comment:
Be good to abstract the type here so it can be reused in the createAnyOfSchema, etc parameter types.

```ts

export interface AnyOfSchema7 extends JSONSchema7 {
  anyOf: JSONSchema7Definition[]
}

export const hasAnyOf = (value: JSONSchema7): value is AnyOfSchema7 => { ... }

// src/yup/schemas/composition/index.ts

export const createAnyOfSchema = (
  [key, value]: [string, AnyOfSchema7],
  jsonSchema: JSONSchema7
): Yup.MixedSchema<string> => { ... }

```